servers/src/test/java/tachyon/worker/block/meta/StorageDirTest.java

```diff
@@ -49,11 +57,101 @@
   public void before() {
     TachyonConf tachyonConf = new TachyonConf();
     mTier = new StorageTier(tachyonConf, 0 /* level */, 1 /* alias */);
-    mDir = new StorageDir(mTier, TEST_DIR_INDEX, TEST_DIR_CAPACITY, TEST_DIR_PATH);
+    mDir = StorageDir.newStorageDir(mTier, TEST_DIR_INDEX, TEST_DIR_CAPACITY, TEST_DIR_PATH);
     mBlockMeta = new BlockMeta(TEST_BLOCK_ID, TEST_BLOCK_SIZE, mDir);
     mTempBlockMeta = new TempBlockMeta(TEST_USER_ID, TEST_TEMP_BLOCK_ID, TEST_BLOCK_SIZE, mDir);
   }
 
+  private StorageDir newStorageDir(File testDir) {
+    return StorageDir.newStorageDir(mTier, TEST_DIR_INDEX, TEST_DIR_CAPACITY,
+        testDir.getAbsolutePath());
+  }
+
+  private File newBlockFile(File dir, String name, int lenBytes) throws IOException {
+    File block = new File(dir, name);
+    block.createNewFile();
+    byte[] data = TestUtils.getIncreasingByteArray(lenBytes);
+    TestUtils.writeBufferToFile(block.getAbsolutePath(), data);
+    return block;
+  }
+
+  private File newTestDir() {
+    File testDir = Files.createTempDir();
+    LOG.debug("new TEST_DIR = {}", testDir.getAbsolutePath());
+    return testDir;
+  }
+
+  @Test
+  public void initializeMetaNoExceptionTest() throws IOException {
+    File testDir = newTestDir();
+
+    int nBlock = 10;
+    long availableBytes = TEST_DIR_CAPACITY;
+    for (int blockId = 0; blockId < nBlock; blockId ++) {
+      int blockSizeBytes = blockId + 1;
+      newBlockFile(testDir, String.valueOf(blockId), blockSizeBytes);
+      availableBytes -= blockSizeBytes;
+    }
+
+    mDir = newStorageDir(testDir);
+    Assert.assertEquals(TEST_DIR_CAPACITY, mDir.getCapacityBytes());
+    Assert.assertEquals(availableBytes, mDir.getAvailableBytes());
+    for (int blockId = 0; blockId < nBlock; blockId ++) {
+      Assert.assertTrue(mDir.hasBlockMeta(blockId));
+    }
+
+    FileUtils.deleteDirectory(testDir);
```

Comment:
If test directory is created by JUNIT, you don't have to manually delete the dir --- one more pro I guess :)